app/app_test.go

```diff
@@ -236,3 +254,376 @@ func TestSpendTx(t *testing.T) {
+	require.Equal(t, expected2, utxo2, "Second UTXO created from 2 different inputs 2 outputs did not get added to the store correctly")
+
+	utxo1 = cc.utxoMapper.GetUTXO(ctx, addrs[3], types.NewPosition(9, 0, 0, 0))
+	require.Nil(t, utxo1, "first UTXO spent in block 10 was not removed correctly from the utxo store")
+
+	utxo2 = cc.utxoMapper.GetUTXO(ctx, addrs[4], types.NewPosition(9, 0, 1, 0))
+	require.Nil(t, utxo2, "second UTXO spent in block 10 was not removed correctly from the utxo store")
+
+	cc.EndBlock(abci.RequestEndBlock{})
+	cc.Commit()
+	cc.BeginBlock(abci.RequestBeginBlock{Header: abci.Header{Height: 11}})
+
+	// merge utxos
+	// Block 11
+	confirmSig1 = CreateConfirmSig(types.NewPosition(10, 0, 0, 0), keys[3], keys[4], true)
+	confirmSig2 = CreateConfirmSig(types.NewPosition(10, 0, 1, 0), keys[3], keys[4], true)
+
+	msg.Blknum1 = 10
+	msg.ConfirmSigs1 = confirmSig1
+	msg.Blknum2 = 10
+	msg.Owner2 = addrs[3]
+	msg.ConfirmSigs2 = confirmSig2
+	msg.Denom1 = 100
+	msg.Newowner2 = common.Address{}
+	msg.Denom2 = 0
+
+	hash = ethcrypto.Keccak256(msg.GetSignBytes())
+	sig1, _ = ethcrypto.Sign(hash, keys[3])
+	tx = types.NewBaseTx(msg, []types.Signature{{
+		Sig: sig1,
+	}, {
+		Sig: sig1,
+	}})
+
+	cres = cc.Check(tx)
+	require.Equal(t, sdk.CodeType(0), sdk.CodeType(cres.Code), cres.Log)
+
+	dres = cc.Deliver(tx)
+	require.Equal(t, sdk.CodeType(0), sdk.CodeType(dres.Code), dres.Log)
+
+	ctx = cc.NewContext(false, abci.Header{})
+
+	utxo1 = cc.utxoMapper.GetUTXO(ctx, addrs[3], types.NewPosition(11, 0, 0, 0))
+	expected1 = types.NewBaseUTXO(addrs[3], [2]common.Address{addrs[3], addrs[3]}, 100, types.NewPosition(11, 0, 0, 0))
+
+	require.Equal(t, expected1, utxo1, "First UTXO created from merge tx did not get added to the store correctly")
+
+	utxo1 = cc.utxoMapper.GetUTXO(ctx, addrs[3], types.NewPosition(10, 0, 0, 0))
+	require.Nil(t, utxo1, "first UTXO spent in block 11 was not removed correctly from the utxo store")
+
+	utxo2 = cc.utxoMapper.GetUTXO(ctx, addrs[3], types.NewPosition(10, 0, 1, 0))
+	require.Nil(t, utxo2, "second UTXO spent in block 11 was not removed correctly from the utxo store")
+}
+
+// Test that several txs can go into a block and that txindex increments correctly
+// Change value of N to increase or decrease txs in the block
+func TestMultiTxBlocks(t *testing.T) {
+	const N = 20
+	// Initialize child chain with deposit
+	cc := newChildChain()
+	var keys [N]*ecdsa.PrivateKey
+	var addrs [N]common.Address
+	var msgs [N]types.SpendMsg
+	var txs [N]sdk.Tx
+
+	for i := 0; i < N; i++ {
+		keys[i], _ = ethcrypto.GenerateKey()
+		addrs[i] = utils.PrivKeyToAddress(keys[i])
+	}
+
+	// Make genesis state with several txs
+	var genUTXOs []GenesisUTXO
+	for i := 0; i < N; i++ {
+		genUTXOs = append(genUTXOs, NewGenesisUTXO(addrs[i].Hex(), "100", [4]string{"0", "0", "0", fmt.Sprintf("%d", i+1)}))
+	}
+
+	genState := GenesisState{
+		UTXOs: genUTXOs,
+	}
+
+	genBytes, err := json.Marshal(genState)
```

Comment:
Rather than doing it this way, refactor InitChain to take an arbitrary number of address arguments